modular_skyrat/modules/bsa_overhaul/code/station_goal.dm

```diff
@@ -0,0 +1,30 @@
+/**
+ * Bluespace artillery goal
+ *
+ * Requires the crew to construct an artillery piece by shift end.
+ */
+
+/datum/station_goal/bluespace_cannon
+	name = "Bluespace Artillery"
+
+/datum/station_goal/bluespace_cannon/get_report()
+	return list(
+		"<blockquote>Our military presence is inadequate in your sector.",
+		"We need you to construct BSA-[rand(1,99)] Artillery position aboard your station.",
+		"",
+		"Base parts are available for shipping via cargo.",
+		"-Nanotrasen Naval Command</blockquote>",
+	).Join("\n")
+
+/datum/station_goal/bluespace_cannon/on_report()
+	//Unlock BSA parts
+	var/datum/supply_pack/engineering/bsa/parts = SSshuttle.supply_packs[/datum/supply_pack/engineering/bsa]
+	parts.special_enabled = TRUE
+
+/datum/station_goal/bluespace_cannon/check_completion()
+	if(..())
+		return TRUE
+	var/obj/machinery/bsa/full/built_cannon = locate()
```

Comment:
there's probably a better way to do this than an `in world` check

Reply:
This is how TG had it, I could make another useless global variable, but this only runs ONCE so it's not actually gonna damage performance